### PR TITLE
chore: add docs to npm directory

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://github.com/cypress-io/cypress/issues/21932

### User facing changelog

N/A

### Additional details

Added some documentation to the `npm` directory about semantic release, the different packages we have, etc.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: npm/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# npm
```
Comment (emilyrohrbough) [inline npm/README.md:1]: Any way we can link this to: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#cypress-and-packages and/or update the contributing guide and link to that section here?

I'd prefer keep this doc & link in contributing doc

File: npm/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+This directory contains packages that are both used internally inside the Cypress monorepo [`packages`](../packages) and also published independently on npm under the Cypress organization using the `@cypress` prefix. For example, `vite-dev-server` is published as `@cypress/vite-dev-server`.
+
+## Release Process
+
+These are automatically released based on [Semantic Version](https://semver.org) commit message prefixes (`feat`, `chore` etc). You can read more about semantic versioning [here](https://semver.org).
+
+This happens when **develop** is merged into **master**. This is usually on a bi-weekly basis, when we do a Cypress release, but can happen out of sync by simply forking from `master` and making a PR directly against `master`.
```
[the remark this answers is not in the document]
Comment (AtofStryker) [reply npm/README.md:9]: I wonder if it makes sense to move that section here and link the reference in the main contributing docs?

File: npm/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+- [webpack-batteries-included-preprocessor](./webpack-batteries-included-preprocessor)
+- [webpack-preprocessor](./webpack-preprocessor)
+
+
+## Misc
+
+Some other packages, mainly used internally or for tooling.
+
+- create-cypress-tests
```
Comment (AtofStryker) [inline npm/README.md:45]: should we link to the repositories here or are some of them private?

File: npm/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lmiller1990) [reply npm/README.md:9]: Ahh I didn't even realize we had this :facepalm:

File: npm/README.md (hunk 3 above)
Comment (lmiller1990) [reply npm/README.md:45]: All the ones here are publicly available!

File: npm/README.md (hunk 1 above)
Comment (lmiller1990) [reply npm/README.md:1]: Done, deleted most of this and linked there.

File: npm/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lmiller1990) [reply npm/README.md]: Done

**Review by AtofStryker — APPROVED** [review]

**Pull request merged**